FtpClient/FtpCli/Data.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+
+namespace FtpCli
+{
+  // The Data class saves session data for the server, user
+  // and commands with arguments, if provided
+  public class Data
+  {
+    string server;
+    string user;
+    string[] commands;
+
+    public Data() {
+      server = "";
+      user = "";
+      commands = null;
+    }
+
+    public void setServer(string server) {
+      this.server = server;
+    }
+
+    public string getServer() {
+      return this.server;
+    }
+
+    public void printServer() {
+      Console.WriteLine(this.server);
+    }
+
+    public void setUser(string user) {
+      this.user = user;
+    }
+
+    public string getUser() {
+      return this.user;
+    }
+
+    public void printUser() {
+      Console.WriteLine(this.user);
+    }
+
+    public void setCommands(string args) {
+      string[] splitArgs = args.Split(" ");
+      this.commands = new string[splitArgs.Length];
+      Array.Copy(splitArgs, 0, this.commands, 0, splitArgs.Length);
+    }
+
+    public string[] getCommands() {
+      return this.commands;
+    }
+
+    public void printCommands() {
+      if (this.commands == null) {
+        Console.WriteLine("no commands");
+        return;
+      }
+
+      if (this.commands[0] == "") {
```

Comment:
can we combine this into an or statement with the block above?

Reply:
You're right, it would still work because of short circuiting if null.